datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/ser/InstantSerializerBase.java

```diff
@@ -136,4 +131,25 @@ protected JsonToken serializationShape(SerializerProvider provider) {
         }
         return JsonToken.VALUE_STRING;
     }
+
+    private String formatValue(T value, SerializerProvider provider)
+    {
+        DateTimeFormatter formatter = null;
+        if (_formatter != null) {
+            formatter = _formatter;
+        } else if (defaultFormat != null) {
+            formatter = defaultFormat;
+        }
+
+        ZoneId zone = provider.getTimeZone().toZoneId();
```

Comment:
Yes it does. The `JacksonJodaDateFormat` class has a `createFormatter(SerializerProvider ctxt)` method, which pretty much does the same (with some extra, because that class has a lot of other features):

```java
    public DateTimeFormatter createFormatter(SerializerProvider ctxt)
    {
        DateTimeFormatter formatter = createFormatterWithLocale(ctxt);
        if (!_explicitTimezone) {
            TimeZone tz = ctxt.getTimeZone();
            if ((tz != null) && !tz.equals(_jdkTimezone)) {
                formatter = formatter.withZone(DateTimeZone.forTimeZone(tz));
            }
        }
        return formatter;
    }
```